src/Models/LagrangianParticleTracking/lagrangian_particle_advection.jl

```diff
@@ -48,27 +55,34 @@ bouncing the particle off the immersed boundary with a coefficient or `restituti
     X = flattened_node((x, y, z), ibg)
 
     # Determine current particle cell
-    fi, fj, fk = fractional_indices(X, ibg.underlying_grid, (c, c, c))
+    fi, fj, fk = fractional_indices(X, ibg.underlying_grid, c, c, c)
     i, j, k = truncate_fractional_indices(fi, fj, fk)
 
-    if immersed_cell(i, j, k, ibg)
-        # Determine whether particle was _previously_ in a non-immersed cell
-        i⁻, j⁻, k⁻ = previous_particle_indices
+    # Determine whether particle was _previously_ in a non-immersed cell
+    i⁻, j⁻, k⁻ = previous_particle_indices
 
-        if !immersed_cell(i⁻, j⁻, k⁻, ibg)
-            # Left-right bounds of the previous, non-immersed cell
-            xᴿ, yᴿ, zᴿ = node(i⁻+1, j⁻+1, k⁻+1, ibg, f, f, f)
-            xᴸ, yᴸ, zᴸ = node(i⁻,   j⁻,   k⁻,   ibg, f, f, f)
+    # Left bounds of the previous cell
+    xᴿ = xnode(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
+    yᴿ = ynode(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
+    zᴿ = znode(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
```

Comment:
```suggestion
    xᴿ = xnode(i⁻ + 1, j⁻, k⁻, ibg, f, f, f)
    yᴿ = ynode(i⁻, j⁻ + 1, k⁻, ibg, f, f, f)
    zᴿ = znode(i⁻, j⁻, k⁻ + 1, ibg, f, f, f)
```
sorry if I am going back and forth, but I am now convinced this is the correct right bounds formulation, the left bounds look good

Reply:
Are you sure about this? The original algorithm had this: 

```julia
xᴿ, yᴿ, zᴿ = node(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
```

The `node()` call is equivalent to three independent calls to `xnode`, `ynode` and `znode` with the same arguments:

https://github.com/CliMA/Oceananigans.jl/blob/0f52a6a00db0cca06785df7ac5658638dd85d5a9/src/Grids/nodes_and_spacings.jl#L35-L37

So to me, the original algorithm should translate to:

```julia
xᴿ = xnode(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
yᴿ = ynode(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
zᴿ = znode(i⁻ + 1, j⁻ + 1, k⁻ + 1, ibg, f, f, f)
```

which is what this PR currently has implemented. Am I missing something?

Or maybe you're saying that the original algorithm was wrong to begin with?